Integrations/integration-LogRhythm.yml

```diff
@@ -718,6 +843,171 @@ script:
         }
     }
 
+    function getHostsByEntityId(dataArgs){
+        var dataObj = {
+            Username: params.Credentials.identifier,
+            Password: params.Credentials.password,
+            EntityId: dataArgs['entity-id']
+        };
+
+        var soapActionURIUgethostbyentity = 'http://www.logrhythm.com/webservices/HostService/GetHostsByEntity';
+        var envelope = fixTemplate(tmplGetHostByEntity, dataObj);
+        var jsResponse = parseResponse(sendRequest(envelope, soapActionURIUgethostbyentity, logRhythmHostAPI));
+
+        var mdRaw = dq(jsResponse, 'Envelope.Body.GetHostsByEntityResponse.GetHostsByEntityResult.HostDataModel')
+
+        if (mdRaw['-i']) {
+            delete mdRaw['-i'];
+        }
+
+        fixDictionaryKeys(mdRaw);
+
+        mdRaw.forEach(function(row){
+            hostsFieldsToRemove.forEach(function(field){
+                delete row[field];
+            });
+        });
+
+        var md = tableToMarkdown('LogRhythm hosts for entity ID: ' + args['entity-id'],  mdRaw);
+
+        var ec = {};
+        ec["LogRhythm.Host(val.ID && val.ID == " + mdRaw['ID'] + ")"] = mdRaw;
+
+        return {
+            Type: entryTypes.note,
+            Contents: jsResponse,
+            ContentsFormat: formats.json,
+            HumanReadable: md,
+            EntryContext: ec
+        };
+
+    }
+
+    function addHost(dataArgs){
+        var dataObj = {
+            Username: params.Credentials.identifier,
+            Password: params.Credentials.password,
+            Hostname: dataArgs['host-name'],
+            EntityId: dataArgs['entity-id'],
+            ShortDesc: dataArgs['short-description'],
+            LongDesc: dataArgs['long-description'],
+            DateUpdated: new Date().toISOString(),
+            RiskThreshold: dataArgs['risk-threshold'],
+            OsType: dataArgs['os-type'],
+            OsVersion: dataArgs['os-version'],
+            ThreatThreshold: dataArgs['threat-threshold']
+        };
+
+        var soapActionURIUaddHost = 'http://www.logrhythm.com/webservices/HostService/AddHost';
+
+        var envelope = fixTemplate(tmplAddHost1, dataObj);
+
+        if(dataArgs['ip-address']){
+            envelope += fixTemplate(tmplAddHostIdentifier, {Type: 'IPAddress', Value: dataArgs['ip-address']});
+        }
+        if(dataArgs['dns-name']){
```

Comment:
can we specify more than one identifier? if not, please use else-if.
if we don't have any identifier, will it cause a problem?

Reply:
yes, its possible to add host with more than one identifier
and its possible to add host with no identifiers